src/Component.js

```diff
@@ -0,0 +1,62 @@
+
+var __extends = (this && this.__extends) || function (d, b) {
```

Comment:
React doesn't seem to like the classes created via that `extend` function as much and chokes on them. To get it to work right I have to add about as much code as that `__extends` function takes up in the first place, so I'm gonna conclude that we may as well keep it using that `__extends` as it works rock solid from the start.